crates/ra_hir/src/ty/tests/data/tuple.txt

```diff
@@ -0,0 +1,17 @@
+[11; 116) '{     ...d"); }': ()
+[21; 22) 'a': (u32, &str)
+[38; 46) '(1, "a")': (u32, &str)
+[39; 40) '1': u32
+[42; 45) '"a"': &str
+[56; 57) 'b': ((u32, &str), [unknown])
+[60; 68) '(a, "b")': ((u32, &str), [unknown])
+[61; 62) 'a': (u32, &str)
+[64; 67) '"b"': [unknown]
+[79; 80) 'c': ([unknown], [unknown])
```

Comment:
@flodiebold Thanks for your quick review.
I'm not sure inferring result after 10 line is right.

Reply:
oh, of course, inferring string literals will only work with #485 :) you could instead add some parameters with declared types and use them instead of literals... so for example
```rust
fn test(x: &str, y: isize) {
    let a: (u32, &str) = (1, "a");
    let b = (a, x);
     let c = (y, x);
    let d = (c, x);
}
```
should be able to infer everything...

Reply:
I see, thanks for your advice.